wordpress-mobile/WordPress-iOS
# Reader: Remove `readerImprovements` feature flag

Closes #22101 

## Description

Removes the `readerImprovements` feature flag. I did attempt to remove some old files such as `OldReaderPostCardCell` but couldn't because it's still being used by the ghost view.

## Testing

To test:
- Launch Jetpack and login
- Navigate to the Reader
- Perform a general smoke test:
  - Browse through different screens and verify everything uses the new designs
  - Tap on actions and verify they work

## Regression Notes
1. Potential unintended areas of impact
Reader

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
